frame/clearing-house/src/lib.rs

```diff
@@ -1639,6 +1679,32 @@ pub mod pallet {
 
 	// Helper functions - validity checks
 	impl<T: Config> Pallet<T> {
+		fn check_oracle_guard_rails(
+			market: &Market<T>,
+			was_mark_index_too_divergent: bool,
+		) -> Result<(), DispatchError> {
+			// Block if mark-index divergence was pushed too far
+			ensure!(
+				!Self::is_mark_index_too_divergent(market)? || was_mark_index_too_divergent,
```

Comment:
Okay I saw in on of the tests that you are indeed looking to let `was_mark_index_too_divergent = true` pass. 
I just didn't understand why, could you please clarify this point?

Reply:
Say, for instance, that the mark price is too above the index. Longs are paying funding to shorts in that case. Then, suppose that ALICE was to close her long, but her position size isn't too big. Then, it might be the case that even after her closing her position, the mark-index divergence is still too large. But we don't wanna block her from closing her position (she is doing what's better for a healthy market after all)

Reply:
I like the short circuit trick btw

Reply:
Got it. Thanks for the explanation!